src/note.rs

```diff
@@ -11,19 +11,35 @@ macro_rules! define_entry_version {
     () => {
         #[used]
         #[unsafe(link_section = ".note.hermit.entry-version")]
-        static ENTRY_VERSION: $crate::_Note = $crate::_Note::entry_version();
+        static ENTRY_VERSION: $crate::_Note<1> = $crate::_Note::entry_version();
+    };
+}
+
+/// Defines the Uhyve interface version in the note section.
+///
+/// This macro must be used in a module that is guaranteed to be linked.
+/// See <https://github.com/rust-lang/rust/issues/99721>.
+#[macro_export]
+macro_rules! define_uhyve_interface_version {
+    ($version:expr) => {
+        #[used]
+        #[unsafe(link_section = ".note.hermit.uhyve-interface-version")]
+        static INTERFACE_VERSION: $crate::_Note<4> = $crate::_Note::uhyveif_version($version);
+    };
+    () => {
+        $crate::define_uhyve_interface_version!(uhyve_interface::UHYVE_INTERFACE_VERSION);
     };
```

Comment:
Can we remove this match? Requiring explicitly providing the value seems much cleaner to me.